sourcecode/www/src/components/ResourceModal/ResourceStats.jsx

```diff
@@ -19,60 +19,88 @@ const ResourceStats = ({ resourceId }) => {
     );
 
-    if (loading || !response) {
-        return <CircularProgress />;
-    }
-    const last7daysViews = response.data.last7daysViews;
+    const dateRangeViews = response?.data?.dateRangeViews || {} ;
     const datasets = [
         {
             key: 'count',
             name: _.capitalize(t('resource_view', { count: 2 })),
-            dataset: last7daysViews,
+            dataset: dateRangeViews,
         },
     ];
 
+    const fromDate = moment(startDate);
+    const toDate = moment(endDate);
+    const diffDays = toDate.diff(fromDate, 'days');
+
+    const handleDateChange = (event) => {
+        setStartDate(event.target.value);
+    };
+    const toDateChange = (event) => {
+        setEndDate(event.target.value);
+    };
+
     return (
         <>
             <Box pb={2}>
-                <strong>{t('S.VIEWS_PAST_7_DAYS')}</strong>
+                <strong>{t('S.VIEWS')}</strong>
             </Box>
-            <ResponsiveContainer width="100%" height={100}>
-                <LineChart
-                    data={fillEmptyDays(
-                        last7daysViews,
-                        moment().subtract(7, 'days').startOf('day'),
-                        moment().endOf('day'),
-                        {
-                            zeroFields: datasets.map((dataset) => dataset.key),
-                        }
-                    )}
-                    margin={{
-                        left: -20,
-                    }}
-                >
-                    <XAxis dataKey="date" />
-                    <YAxis allowDecimals={false} />
-                    <Tooltip />
-                    {datasets.map((dataset) => (
-                        <Line
-                            key={dataset.name}
-                            dot={false}
-                            name={dataset.name}
-                            type="monotone"
-                            dataKey={dataset.key}
-                            stroke={dataset.color}
-                            yAxisId={0}
-                            isAnimationActive={false}
-                        />
-                    ))}
-                </LineChart>
-            </ResponsiveContainer>
+
+            <Box pb={2}>
+            From: <input type="date" id="from" name="from" max={endDate} onChange={handleDateChange} value={moment(startDate).format('YYYY-MM-DD')} /> {' '}
+            To: <input type="date" id="to" name="to" min={moment(startDate).format('YYYY-MM-DD')} max={moment(new Date()).format('YYYY-MM-DD')} onChange={toDateChange} value={moment(endDate).format('YYYY-MM-DD')}/>
+            </Box>
+
+            {loading || !response ?
+                <CircularProgress/>
+                :
+                <ResponsiveContainer width="100%" height={100}>
+                    <LineChart
+                        data={fillEmptyDays(
+                            dateRangeViews,
+                            moment().subtract(diffDays, 'days').startOf('day'),
```

Comment:
The second and third argument to `fillEmptyDays()` are not correct, as this code operates on the current day, but both the start and end date are now arbitrary.